### DISCUSSION TITLE
Stabilité et fiabilité des API Enedis vis à vis du consentement

### DISCUSSION THREAD

**Discussion by ThibBal** [bokub/conso-api#67] [Général]
Hello ! 

Une petite question de la part d'une équipe qui cherche à implémenter une solution à l'aide de Conso API : j'ai remarqué que le service de consentement d'Enedis et de récupération du fameux token était actuellement cassé et cela depuis plusieurs semaines... sans une grande visibilité sur quand il sera de nouveau opérationnel.

Mes questions sont donc : est-ce que cela arrive "souvent" ? Ce service est-il fiable pour un usage professionnel de suivi d'une consommation ? Sinon, faut-il plutôt passer par une autre solution ?

Merci d'avance de vos réponses et de tout élément qui permettra d'enrichir notre réflexion !

**Comment by bokub** [comment]
Salut !
Oui ça arrive souvent, et non ça n'est pas fiable du tout.

Je ne baserais pas un business là-dessus, mais je crois que les professionnels plus "gros" (style total ou autres fournisseurs d'énergie) passent par d'autres api qui sont moins souvent cassées.

**Comment by ThibBal** [comment]
Est-ce que par hasard tu connaitrais d'autres API sur lesquels les "gros" se basent :) ? 

Merci de ton retour !

**Reply by bokub** [reply]
Non désolé

**Comment by crapulas-cb6** [comment]
@bokub Sur la page Enedis, il semble que les URL des API aient changé

Dès le mois de Novembre 2021, nous avons mis en service la nouvelle plateforme d’API management d’Enedis.

Dans ce contexte, de nouvelles versions des API Authorize, Grid_connection_simulation, Metering et Customers ont vu le jour au fur et à mesure de la migration des API sur cette nouvelle plateforme. Les impacts utilisateurs ont été limités au maximum, mais certaines modifications seront nécessaires pour consommer ces nouvelles APIs dont vous trouverez les détails ci-dessous.

Les versions précédentes des API restent bien sûr fonctionnelles sur l’ancienne plateforme jusqu’au mois de Mars 2023. Passé cette date, seules les API de la nouvelle plateforme seront maintenues.

https://datahub-enedis.fr/migrer-sur-la-nouvelle-plateforme/

Ancienne plateforme : /v1/oauth2/…
Nouvelle plateforme : /oauth2/v3/…

J'ai eu le service client Enedis en ligne mais ils ne savent pas. Ils vont remonter cela à leur service informatique. En revanche, l'URL vers laquelle tu nous renvoies pour l'authentification et la demande de consentement semble être sur l'ancienne plateforme. C'est le même probleme pour MyElectricalData.fr

https://mon-compte-particulier.enedis.fr/dataconnect/v1/oauth2/authorize?client_id...

**Reply by bokub** [reply]
Merci @crapulas-cb6 pour l'info mais si tu lis le README tu verras que c'est justement sur ces nouvelles API que j'ai construit Conso API:

![image](https://github.com/user-attachments/assets/b67f1a3d-b138-4050-8bd8-91c292a387d8)

Effectivement, j'utilise toujours `/v1/oauth2/` parce que c'est toujours l'URL pour l'Authorization, alors qu'avant c'était l'URL pour l'Authorization ET le Jeton (qui lui a changé d'URL). Je t'accorde que leur article n'est pas très clair mais voici la documentation:

|Authorization|Jeton|
|---|---|
| ![image](https://github.com/user-attachments/assets/7941411a-355d-4ea2-9cda-795ed4e3ec5c)| ![image](https://github.com/user-attachments/assets/a6876eb5-b4ad-4d3d-851a-02e70b08e377)|

**Reply by bokub** [reply]
> J'ai eu le service client Enedis en ligne mais ils ne savent pas. Ils vont remonter cela à leur service informatique

ça ne sert à rien de les contacter à nouveau, ils ont déjà reçu plein d'e-mails, cf #62

**Reply by crapulas-cb6** [reply]
> > J'ai eu le service client Enedis en ligne mais ils ne savent pas. Ils vont remonter cela à leur service informatique
> 
> ça ne sert à rien de les contacter à nouveau, ils ont déjà reçu plein d'e-mails, cf #62

J'avais ouvert un ticket il y a quelques semaines et viennent de me rappeler :)

**Reply by bokub** [reply]
Ah top ! Dans ce cas je retire ce que j'ai dit ça doit servir à quelque chose de les harceler. Ils t'ont dit quoi ?

**Reply by dpobel** [reply]
@crapulas-cb6 du coup le consentement fonctionne pour toi ?

**Reply by crapulas-cb6** [reply]
> @crapulas-cb6 du coup le consentement fonctionne pour toi ?

Non…
